File: app/src/main/java/com/zetapush/testsdk/KeyValueTokenStorage.java
package com.zetapush.testsdk;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

public class KeyValueTokenStorage implements StorageTokenInterface {

    // Variable
    private final Context context;
    private final String keyToken;

    // Constructor
    public KeyValueTokenStorage(Context context, String keyToken) {
        this.context = context;
        this.keyToken = keyToken;
    }

    @Override
    public void saveToken(String token) {
        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putString(keyToken, token);
        editor.commit();
    }

    @Override
    public String getToken() {
        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
        return sharedPreferences.getString(keyToken, null);
    }

    @Override
    public void clearToken() {
        this.saveToken(null);
    }
}

File: android-library/src/main/java/com/zetapush/library/KeyValueTokenStorage.kt
package com.zetapush.library

import android.content.Context
import android.preference.PreferenceManager

class KeyValueTokenStorage(
    private val context: Context,
    private val keyToken: String
) : StorageTokenInterface {

    override fun saveToken(token: String?) {
        val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
        val editor = sharedPreferences.edit()
        editor.putString(keyToken, token)
        editor.apply()
    }

    override fun getToken(): String? {
        val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
        return sharedPreferences.getString(keyToken, null)
    }

    override fun clearToken() {
        this.saveToken(null)
    }
}